static/js/message_list_view.js

```diff
@@ -326,8 +326,35 @@ export class MessageListView {
 
         message_container.is_hidden = is_hidden;
         // Make sure the right thing happens if the message was edited to mention us.
-        message_container.contains_mention = message_container.msg.mentioned && !is_hidden;
+        if (!is_hidden && message_container.msg.mentioned) {
+            // Currently the API does not differentiate between a group mention and
+            // a user mention. For now, we parse the markdown to see if the message
+            // mentions the user.
+            let is_user_mention = false;
+            const $msg = $(message_container.msg.content);
+            $msg.find(".user-mention:not(.silent)").each(function () {
+                const user_id = rendered_markdown.get_user_id_for_mention_button(this);
+                if (user_id === "*") {
+                    return;
+                }
+                if (people.is_my_user_id(user_id)) {
+                    is_user_mention = true;
+                }
+            });
 
+            // If a message includes a user mention, even alongside a group/wildcard mention,
+            // color it as a user mention.
+            if (message_container.msg.mentioned_me_directly && is_user_mention) {
+                message_container.contains_direct_user_mention = true;
+                message_container.contains_group_mention = false;
+            } else {
+                message_container.contains_group_mention = true;
+                message_container.contains_direct_user_mention = false;
+            }
```

Comment:
So to check if there's a group mention, I'd have expected to parse the message looking for `.user-group-mention`. I think it's fine to not do that if you can infer it from the fact that we don't have a different mention type but do have a mention, but you absolutely should add a comment explaining the logic, since it's likely the kind of subtle thing that could break when adding a future mention type.

Also worth spelling out is how this logic is handling wildcard mentions -- it looks like those are treated as group mentions for this purpose because `is_user_mention` will be false? 

But honestly it might be a lot more readable and extensible to just adjust your selector to find the group mentions too, and compute variables `has_wildcard_mention`, `has_user_mention`, `has_group_mention` and then make decisions based on that.

Reply:
Is there a better term I could use to describe usergroup + wildcard mentions together?

Reply:
I simplified this a bit in a different way where I just set one classname for the message (since we're only going to give it one colour). That feels more readable and extensible to me, and simpler than computing those variables. Let me know what you think!